modules/dublincore/src/test/java/org/opencastproject/metadata/dublincore/DublinCoreXmlFormatTest.java

```diff
@@ -97,4 +100,59 @@ public void readFromNode() throws Exception {
     assertNotNull(dc);
     assertEquals("10.0000/5820", DublinCores.mkOpencastEpisode(dc).getDcIdentifier().get());
   }
+
+  @Test
+  public void readFromFileWithEmpty() throws Exception {
+    String emtpy = "";
+    boolean marshalEmpty = true;
+    assertEquals(
+            emtpy,
+            DublinCoreXmlFormat
+            .read(IoSupport.classPathResourceAsFile("/dublincore.xml").get(), marshalEmpty)
+            .getFirst(DublinCore.PROPERTY_SPATIAL));
+    assertNull(
+            DublinCoreXmlFormat
+            .read(IoSupport.classPathResourceAsFile("/dublincore.xml").get(), !marshalEmpty)
+            .getFirst(DublinCore.PROPERTY_SPATIAL));
+  }
+
+  @Test
+  public void readWithEmptyAndMergeTest() throws Exception {
+    String emtpy = "";
+    String delimiter = "|";
+    String irishTitle = "Talamh agus Fásra: Príomh-imreoirí ar an Radharc Aeráide";
+    String irishLang = "ga";
+    String germanLang = "de";
+    String fooId = "197011";
+    EName eNamepropertyFooId = new EName("http://foo.org/metadata", "id");
+    boolean marshalEmpty = true;
+
+    DublinCoreCatalog catalogWithEmpty = DublinCoreXmlFormat
+            .read(IoSupport.classPathResourceAsFile("/dublincore.xml").get(), marshalEmpty);
+    DublinCoreCatalog catalogNoEmpty = DublinCoreXmlFormat
+            .read(IoSupport.classPathResourceAsFile("/dublincore.xml").get(), !marshalEmpty);
+
+    DublinCoreCatalog catalogToMerge = DublinCoreXmlFormat
+            .read(IoSupport.classPathResourceAsFile("/dublincore-extended.xml").get(), marshalEmpty);
+    DublinCoreCatalog merged = DublinCoreXmlFormat.merge(catalogToMerge, catalogNoEmpty);
+
+    // Loading with empty and not empty
+    assertEquals(emtpy,catalogWithEmpty.getFirst(DublinCore.PROPERTY_SPATIAL));
+    assertNull(catalogNoEmpty.getFirst(DublinCore.PROPERTY_SPATIAL));
+    assertEquals("lab2", catalogToMerge.getFirst(DublinCore.PROPERTY_SPATIAL));
+
+    // Merging same catalog with extra empty values does not affect the original
+    assertEquals(catalogNoEmpty, DublinCoreXmlFormat.merge(catalogWithEmpty, catalogNoEmpty));
+    assertEquals(irishTitle, catalogNoEmpty.getAsText(DublinCore.PROPERTY_TITLE, irishLang, delimiter));
+    assertNull(catalogNoEmpty.getFirst(DublinCore.PROPERTY_SPATIAL));
+
+    // Merging catalog with different values
+    assertEquals("lab2", merged.getFirst(DublinCore.PROPERTY_SPATIAL));
+    assertNull(merged.getAsText(DublinCore.PROPERTY_TITLE, irishLang, delimiter));
+    assertNotNull(merged.getAsText(DublinCore.PROPERTY_TITLE, germanLang, delimiter));
+    assertTrue("Property foo:id should be in the list of known properties.",
+            merged.getProperties().contains(eNamepropertyFooId));
+    assertEquals(fooId, merged.getFirstVal(eNamepropertyFooId).getValue());
+
+  }
```

Comment:
I had a bit of trouble understanding this test on a first glance, so I rearranged some of the lines, gonna leave this here as a suggestion:
```
    DublinCoreCatalog catalogWithEmpty = DublinCoreXmlFormat
            .read(IoSupport.classPathResourceAsFile("/dublincore.xml").get(), marshalEmpty);
    DublinCoreCatalog catalogNoEmpty = DublinCoreXmlFormat
            .read(IoSupport.classPathResourceAsFile("/dublincore.xml").get(), !marshalEmpty);

    // Loading with empty and not empty
    assertEquals(emtpy,catalogWithEmpty.getFirst(DublinCore.PROPERTY_SPATIAL));
    assertNull(catalogNoEmpty.getFirst(DublinCore.PROPERTY_SPATIAL));

    DublinCoreCatalog mergedEmptyNoEmpty = DublinCoreXmlFormat.merge(catalogWithEmpty, catalogNoEmpty);

    // Extra empty values have no effect on merge, original is not affected
    assertEquals(catalogNoEmpty, mergedEmptyNoEmpty);
    assertEquals(irishTitle, catalogNoEmpty.getAsText(DublinCore.PROPERTY_TITLE, irishLang, delimiter));
    assertNull(catalogNoEmpty.getFirst(DublinCore.PROPERTY_SPATIAL));


    DublinCoreCatalog catalogToMerge = DublinCoreXmlFormat
            .read(IoSupport.classPathResourceAsFile("/dublincore-extended.xml").get(), marshalEmpty);

    assertEquals("lab2", catalogToMerge.getFirst(DublinCore.PROPERTY_SPATIAL));

    DublinCoreCatalog mergedToMergeNoEmpty = DublinCoreXmlFormat.merge(catalogToMerge, catalogNoEmpty);

    // Merging catalog with different values removes empty values and adds or updates non-empty values
    assertEquals("lab2", mergedToMergeNoEmpty.getFirst(DublinCore.PROPERTY_SPATIAL));
    assertNull(mergedToMergeNoEmpty.getAsText(DublinCore.PROPERTY_TITLE, irishLang, delimiter));
    assertNotNull(mergedToMergeNoEmpty.getAsText(DublinCore.PROPERTY_TITLE, germanLang, delimiter));
    assertTrue("Property foo:id should be in the list of known properties.",
            mergedToMergeNoEmpty.getProperties().contains(eNamepropertyFooId));
    assertEquals(fooId, mergedToMergeNoEmpty.getFirstVal(eNamepropertyFooId).getValue());
```

Reply:
@KatrinIhler thank you for looking at this! I'll update those tests.